dataproc/create_hail_cluster.ipynb

```diff
@@ -0,0 +1,865 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "78203c21-40bf-4b8a-b2f5-1d352fa1319e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "SERVICE_ACCOUNT = os.getenv('GOOGLE_SERVICE_ACCOUNT_EMAIL')\n",
+    "\n",
+    "SERVICE_ACCOUNT"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "49537cb3-20b4-4135-b316-efb95a995172",
+   "metadata": {},
+   "source": [
+    "Obtain the user name so that it can become part of the Hail cluster name and staging bucket. This is useful when people collaborate in Enterprise Terra workspaces and want to differentiate their clusters from each other."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "47735c99-12d4-47c2-b2ae-2e21da4f73cf",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "USER = os.getenv('TERRA_USER_EMAIL').split('@')[0].replace('.', '-')\n",
+    "\n",
+    "USER"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1133b444-a427-4a72-a92b-0acb2c146844",
+   "metadata": {},
+   "source": [
+    "## Check the Enterprise Terra reference to the Dataproc staging bucket\n",
+    "\n",
+    "Create the user's Dataproc staging bucket, if it does not yet exist."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1e9af1d8-64f4-4d1a-ba32-e5eb4eff6565",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!terra resource resolve --name dataproc_staging_{USER} || terra resource create gcs-bucket --name=dataproc_staging_{USER} \\\n",
+    "    --bucket-name={PROJECT}-dataproc-staging-{USER} \\\n",
+    "    --cloning=COPY_NOTHING \\\n",
+    "    --description=\"Bucket for {USER} Dataproc staging files. See https://cloud.google.com/dataproc/docs/concepts/configuring-clusters/staging-bucket\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cdb76aa0-e806-4689-9be9-2ba5a65c2c0d",
+   "metadata": {},
+   "source": [
+    "Resolve the reference to the staging bucket so that we can use it in subsequent commands."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "de31fc7e-b743-4317-8bd0-bfdb0945bbd1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "STAGING_BUCKET_CMD_OUTPUT = !terra resolve --name=dataproc_staging_{USER}\n",
+    "STAGING_BUCKET = STAGING_BUCKET_CMD_OUTPUT[0]\n",
+    "STAGING_BUCKET"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6376d1d7-1016-47b1-8f86-ecb021d147af",
+   "metadata": {},
+   "source": [
+    "## Check the Enterprise Terra reference to the Dataproc temp bucket\n",
+    "\n",
+    "<div class=\"alert alert-block alert-info\">\n",
+    "<b>Note:</b> If you did not reach this notebook by duplicating the \"Hail Demo\" workspace, then you will need to first run  <a href=\"https://github.com/DataBiosphere/terra-axon-examples/blob/main/workspace_setup.ipynb\">workspace_setup.ipynb</a>, which is in the parent directory of this notebook.  \n",
```

Comment:
`workspace_setup.ipynb` no longer creates resources that are cloned by default, so people do need to run it once per workspace